lib/kafo/configuration.rb

```diff
@@ -151,17 +171,47 @@ def params
       @params ||= modules.map(&:params).flatten
     end
 
-    def preset_parameters
+    def param(mod, name)
+      params.detect { |p| p.name == name && p.module.name == mod }
+    end
+
+    def preset_defaults_from_puppet
       # set values based on default_values
       params.each do |param|
         param.set_default(params_default_values)
       end
+    end
 
+    def preset_defaults_from_yaml
       # set values based on YAML
       params.each do |param|
         param.set_value_by_config(self)
       end
-      params
+    end
+
+    def preset_defaults_from_other_config(other_config)
+      params_changed(other_config).each do |par|
+        param(par.module.class_name, par.name).value = other_config.param(par.module.class_name, par.name).value
+      end
+    end
+
+    def params_changed(old_config)
+      # finds params that had different value in the old config
+      params.select do |par|
+        old_param = old_config.param(par.module.class_name, par.name)
+        old_param && old_param.value != par.value
+      end
+    end
+
+    def params_missing(old_config)
+      # finds params that are present but will be missing in the new config
+      old_config.params.select do |par|
+        param(par.module.class_name, par.name).nil?
```

Comment:
This will cover situation, when module present in previous is missing from new scenario. All affected parameters from previous will be listed. E.g. I had proxy installed and wanted to reinstall full foreman scenario. There were some proxy plugins missing from the new scenario and this warned me.